pennylane/operation.py

```diff
@@ -447,6 +447,10 @@ def check_domain(self, p, flattened=False):
         Returns:
             Number, array, Variable: p
         """
+        # If parameter is a NumPy scalar, convert it into a Python scalar.
+        if isinstance(p, np.ndarray) and p.ndim == 0:
+            p = p.item()
```

Comment:
This change allows NumPy scalars to be used within gate arguments, e.g.
```python
>>> a = np.array(0.5)
>>> circuit(a)
```

PennyLane core already works fine with NumPy scalars, it was just a matter of input validation blocking it. This change is required for the user to mark scalars as non-differentiable when using Autograd:

```python
>>> a = np.array(0.5, requires_grad=False)
>>> circuit(a)
```

and mirrors how PyTorch and TF work (e.g., `tf.Variable(0.5)`).